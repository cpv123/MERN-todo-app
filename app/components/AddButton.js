import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const AddButton = (props) => {
    return (
        <Button
            style={{ 'marginTop': 20 }} 
            color="primary"
            variant="fab"
            onClick={props.onClick}
            disabled={props.disabled} >
            <AddIcon />
        </Button>
    )
}

AddButton.propTypes = {
    color: PropTypes.string,
    variant: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired
}

export default AddButton;