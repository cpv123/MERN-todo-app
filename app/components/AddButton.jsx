import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const AddButton = (props) => {
  const { onClick, disabled } = props;
  return (
    <Button
      style={{ marginTop: 20 }}
      color="primary"
      variant="fab"
      onClick={onClick}
      disabled={disabled}
    >
      <AddIcon />
    </Button>
  );
};

AddButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
};

export default AddButton;
