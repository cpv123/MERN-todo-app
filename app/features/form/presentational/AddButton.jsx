import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const AddButton = ({ onClick, disabled }) => (
  <Button
    style={{ marginTop: 20, marginLeft: 40 }}
    color="primary"
    variant="fab"
    onClick={onClick}
    disabled={disabled}
  >
    <AddIcon />
  </Button>
);

AddButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
};

export default AddButton;
