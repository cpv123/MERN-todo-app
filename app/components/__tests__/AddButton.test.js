/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import AddButton from '../AddButton.jsx';

describe('the AddButton component', () => {
  it('renders correctly', () => {
    const mockProps = { onClick: jest.fn() };
    const wrapper = shallow(<AddButton {...mockProps}/>);
    expect(wrapper).toMatchSnapshot();
  });
});