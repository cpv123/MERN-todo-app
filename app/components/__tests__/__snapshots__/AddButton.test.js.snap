// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the AddButton component renders correctly 1`] = `
<WithStyles(Button)
  color="primary"
  onClick={[MockFunction]}
  style={
    Object {
      "marginLeft": 40,
      "marginTop": 20,
    }
  }
  variant="fab"
>
  <pure(Add) />
</WithStyles(Button)>
`;
